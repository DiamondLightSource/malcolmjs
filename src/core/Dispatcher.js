/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
