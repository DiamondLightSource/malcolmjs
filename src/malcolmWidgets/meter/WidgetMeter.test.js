import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import WidgetMeter from './WidgetMeter.component';

describe('WidgetMeter', () => {
  let shallow;

  const theme = {
    palette: {
      primary: {
        main: '#00f',
      },
      secondary: {
        main: '#0f0',
      },
      background: {
        paper: '#424242',
      },
      divider: '#fff',
      Error: '#f00',
    },
  };

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <WidgetMeter value={50} theme={theme} limits={{ limitHigh: 100 }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('displays out-of-bounds correctly', () => {
    const wrapper = shallow(
      <WidgetMeter value={150} theme={theme} limits={{ limitHigh: 100 }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
