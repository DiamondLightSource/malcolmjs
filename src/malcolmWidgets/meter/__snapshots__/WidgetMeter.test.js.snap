// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetMeter displays out-of-bounds correctly 1`] = `
<div
  className="WidgetMeter-div-1"
>
  <div
    style={
      Object {
        "backgroundImage": "linear-gradient(to right, transparent, #f00)",
        "gridColumn": "1",
        "gridRow": "1",
        "left": "0px",
        "minWidth": "100%",
        "position": "relative",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#424242",
        "borderLeft": "2px solid #f00",
        "gridColumn": "1",
        "gridRow": "1",
        "left": "calc(100%)",
        "maxWidth": "0%",
        "minWidth": "0%",
        "position": "relative",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#424242",
        "borderLeft": "2px solid #f00",
        "gridColumn": "1",
        "gridRow": "1",
        "left": "calc(100%)",
        "maxWidth": "0%",
        "minWidth": "0%",
        "position": "relative",
      }
    }
  />
  <div
    style={
      Object {
        "gridColumn": "1",
        "gridRow": "1",
        "position": "relative",
      }
    }
  >
    <WithStyles(WidgetTextInput)
      Pending={false}
      Units="/ 100"
      alwaysContrastText={true}
      localState={
        Object {
          "set": [Function],
        }
      }
    />
  </div>
</div>
`;

exports[`WidgetMeter matches snapshot 1`] = `
<div
  className="WidgetMeter-div-1"
>
  <div
    style={
      Object {
        "backgroundImage": "linear-gradient(to right, transparent, #00f)",
        "gridColumn": "1",
        "gridRow": "1",
        "left": "0px",
        "minWidth": "100%",
        "position": "relative",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#424242",
        "borderLeft": "2px solid #00f",
        "gridColumn": "1",
        "gridRow": "1",
        "left": "calc(50%)",
        "maxWidth": "50%",
        "minWidth": "50%",
        "position": "relative",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#424242",
        "borderLeft": "2px solid #00f",
        "gridColumn": "1",
        "gridRow": "1",
        "left": "calc(50%)",
        "maxWidth": "50%",
        "minWidth": "50%",
        "position": "relative",
      }
    }
  />
  <div
    style={
      Object {
        "gridColumn": "1",
        "gridRow": "1",
        "position": "relative",
      }
    }
  >
    <WithStyles(WidgetTextInput)
      Pending={false}
      Units="/ 100"
      alwaysContrastText={true}
      localState={
        Object {
          "set": [Function],
        }
      }
    />
  </div>
</div>
`;
